### PR TITLE
feat: add GetProtocol and SetProtocol api for ResponseHeader

### PR DESCRIPTION
**Pull request by wzekin**
#### What type of PR is this?

feat
#### Check the PR title.

- [x] This PR title match the format: `<type>(optional scope): <description>`.
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io).

#### (Optional) Translate the PR title into Chinese.
为 ResponseHeader 添加 GetProtocol 和 SetProtocol 接口

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: hertz client's Response lacks a quick and easy way to determine the HTTP version of the Response, so adding `SetProtocol` and `GetProtocol` allows the user to determine the HTTP version of the Response
zh(optional): hertz client 的 Response 缺乏一种快速简便的方式去判断这个 Response 的 HTTP 版本，所以添加`SetProtocol` 和`GetProtocol`让用户可以判断 Response 的 HTTP 版本

### CODE REVIEW

File: pkg/protocol/header.go [hunk 1]
```diff
@@ -112,6 +112,7 @@ type ResponseHeader struct {
 
 	disableNormalizing   bool
 	noHTTP11             bool
```
Comment (Duslia) [inline pkg/protocol/header.go:114]: Is noHTTP1 useless?

File: pkg/protocol/http1/resp/header.go [hunk 2]
```diff
@@ -239,7 +239,15 @@ func parseFirstLine(h *protocol.ResponseHeader, buf []byte) (int, error) {
 	if n < 0 {
 		return 0, fmt.Errorf("cannot find whitespace in the first line of response %q", buf)
 	}
-	h.SetNoHTTP11(!bytes.Equal(b[:n], bytestr.StrHTTP11))
+
+	isHTTP11 := bytes.Equal(b[:n], bytestr.StrHTTP11)
+	h.SetNoHTTP11(!isHTTP11)
+	if !isHTTP11 {
+		h.SetProtocol(consts.HTTP10)
```
Comment (welkeyever) [inline pkg/protocol/http1/resp/header.go:246]: better go with lazy loading?

File: pkg/protocol/header.go [hunk 3]
```diff
@@ -173,8 +173,16 @@ func (h *ResponseHeader) PeekArgBytes(key []byte) []byte {
...
 }
 
+// Deprecated: Use ResponseHeader.SetProtocol(consts.HTTP11) instead
+//
+//	Now SetNoHTTP11(true) equal to SetProtocol(consts.HTTP10)
+//		SetNoHTTP11(false) equal to SetProtocol(consts.HTTP11)
 func (h *ResponseHeader) SetNoHTTP11(b bool) {
-	h.noHTTP11 = b
+	if b {
+		h.protocol = consts.HTTP10
```
Comment (welkeyever) [inline pkg/protocol/header.go:182]: 增加一个return去掉后面的else吧

File: pkg/protocol/header.go [hunk 4]
```diff
@@ -300,8 +307,16 @@ func (h *RequestHeader) GetProtocol() string {
...
 }
 
+// Deprecated: Use RequestHeader.SetProtocol(consts.HTTP11) instead
+//
+//	Now SetNoHTTP11(true) equal to SetProtocol(consts.HTTP10)
+//		SetNoHTTP11(false) equal to SetProtocol(consts.HTTP11)
 func (h *RequestHeader) SetNoHTTP11(b bool) {
-	h.noHTTP11 = b
+	if b {
+		h.protocol = consts.HTTP10
```
Comment (welkeyever) [inline pkg/protocol/header.go:316]: AA

File: pkg/protocol/header.go (hunk 3 above)
Comment (wzekin) [reply pkg/protocol/header.go:182]: done

File: pkg/protocol/header.go (hunk 1 above)
Comment (wzekin) [reply pkg/protocol/header.go]: yeap, I delete noHTTP1 field now

**Review by welkeyever — APPROVED** [review]

**Review by li-jin-gou — APPROVED** [review]

**Review by FGYFFFF — APPROVED** [review]

**Pull request merged**